### PR TITLE
Implement sessions endpoint

### PR DESCRIPTION
**Pull request by ypaq**
For consistency we implement the v3 API sessions endpoint which is equivalent to the v2 version. 

One minor difference between v3 and v2 is that the channel for which the session is created must exist, otherwise we return a `422 Unprocessable Entity`.

### CODE REVIEW

File: doc/schema/schema.json [hunk 1]
```diff
@@ -505,6 +505,52 @@
...
+              "tail": {
+                "description": "if present with any value, start a live tail session",
+                "type": [
+                  "boolean",
+                  "null"
+                ]
+              }
+            },
+            "example": {
+              "channel_id": "d23d5262-d1f0-46f0-93c5-3781b48a258d",
```
Comment (cdwort) [inline doc/schema/schema.json]: slightly better example for `channel_id` would be `app-d23...`.

**Comment by cdwort** [comment]
Taking this review from Pavel since I'm slightly more confident in erlang. 

This looks like it's exactly what we need. Thanks!

I did comment that the channel_id example isn't exactly the format we'll be using, but that's not a big deal.

For my own edification, how do we terminate these sessions from our end? Is there a default session close after a few hours?

**Comment by ypaq** [comment]
Thanks @cdwort! Updated the example. 

The session creation gives you a URL from where you can fetch the logs. For creation we store the data in redis with a timeout which defaults to 360 seconds (see [logplex_session:expiry()](https://github.com/heroku/logplex/blob/master/src/logplex_session.erl#L83)).

Fetching the logs is still handled by the v2 API but you don't have to worry about that as we provide you with the URL.

**Comment by rwz** [comment]
LGTM (entirely based on @cdwort's judgement since I can't really read this) :)

**Comment by cdwort** [comment]
Ah, okay @ypaq, so there's no external session termination -- the logs just stop showing up at some point?

**Comment by rwz** [comment]
:+1: please let us know when this is available in staging

**Comment by rwz** [comment]
Is there a production deployment we could test this against?

**Pull request merged**